cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerUpgradeParcelDownloadListenerTask.java

```diff
@@ -37,14 +48,15 @@ public ClouderaManagerUpgradeParcelDownloadListenerTask(ClouderaManagerApiPojoFa
     protected boolean doStatusCheck(ClouderaManagerCommandPollerObject pollerObject) throws ApiException {
 
         ApiParcel apiParcel = getApiParcel(pollerObject);
-        String parcelStage = apiParcel.getStage();
-
-        if (!ParcelStatus.DOWNLOADED.name().equals(parcelStage)
-                && !ParcelStatus.DISTRIBUTED.name().equals(parcelStage)
-                && !ParcelStatus.DISTRIBUTING.name().equals(parcelStage)
-                && !ParcelStatus.ACTIVATED.name().equals(parcelStage)
-                && !ParcelStatus.ACTIVATING.name().equals(parcelStage)) {
-            LOGGER.warn("Expected parcel status is {}, received status is: {}", ParcelStatus.DOWNLOADED.name(), parcelStage);
+
+        List<String> errors = apiParcel.getState().getErrors();
```

Comment:
this logic should be included in `ClouderaManagerUpgradeParcelDistributeListenerTask` and `ClouderaManagerSingleParcelActivationListenerTask`, too, could you pls do that or would you prefer a follow-up PR?